base/base.jl

```diff
@@ -34,6 +34,11 @@ type MethodError <: Exception
     args
 end
 
+type AmbiguousError <: Exception
```

Comment:
Would it be better to be more long-winded: `AmbiguousMethodError <: MethodError` ? (and `NoMethodError <: MethodError`.)